File: monty_hall.py
from __future__ import division
from random import randint

#Number of games and game generation
ngames = 10000

prize = [randint(1,3)*x for x in ngames*[1]]
player = [randint(1,3)*x for x in ngames*[1]]

# Evaluating success
default = 0
monty = 0

for game in range(ngames):
    # Keeping decision
    if prize[game] == player[game]:
        default = default + 1
    # Taking chance with Monty
    if prize[game] == player[game]:
        # Switching means losing
        pass
    # Switching means winning
    else:
        monty = monty + 1
print 'Probability of winning (own door)', default/ngames
print 'Probability of winning (switching doors)', monty/ngames

File: monty_hall.jl
function monty_sim(ngames)
  no_switch = 0
  switch = 0
  for game in 1:ngames
    player = rand(1:3)
    prize = rand(1:3)
    if player == prize
      no_switch += 1
    else
      switch += 1
    end
  end
  println("Probability of winning (own door): $(no_switch/ngames)")
  println("Probability of winning (switching door): $(switch/ngames)")
end

monty_sim(30000000)
